manim/utils/color.py

```diff
@@ -21,350 +26,436 @@
-RED_D: str = "#E65A4C"
-RED_E: str = "#CF5044"
-PURE_RED: str = "#FF0000"
-RED: str = "#FC6255"
-MAROON_A: str = "#ECABC1"
-MAROON_B: str = "#EC92AB"
-MAROON_C: str = "#C55F73"
-MAROON_D: str = "#A24D61"
-MAROON_E: str = "#94424F"
-MAROON: str = "#C55F73"
-PURPLE_A: str = "#CAA3E8"
-PURPLE_B: str = "#B189C6"
-PURPLE_C: str = "#9A72AC"
-PURPLE_D: str = "#715582"
-PURPLE_E: str = "#644172"
-PURPLE: str = "#9A72AC"
-PINK: str = "#D147BD"
-LIGHT_PINK: str = "#DC75CD"
-ORANGE: str = "#FF862F"
-LIGHT_BROWN: str = "#CD853F"
-DARK_BROWN: str = "#8B4513"
-GRAY_BROWN: str = "#736357"
-GREY_BROWN: str = "#736357"
+                return lines
+
+            other_colors = (
+                "pink",
+                "light_pink",
+                "orange",
+                "light_brown",
+                "dark_brown",
+                "gray_brown",
+            )
+
+            other_lines = named_lines_group(
+                3.2,
+                other_colors,
+                other_colors,
+                [BLACK] * 4 + [WHITE] * 2,
+                0,
+            )
+
+            gray_lines = named_lines_group(
+                6.6,
+                ["white"] + subnames("gray") + ["black"],
+                [
+                    "white",
+                    "lighter_gray / gray_a",
+                    "light_gray / gray_b",
+                    "gray / gray_c",
+                    "dark_gray / gray_d",
+                    "darker_gray / gray_e",
+                    "black",
+                ],
+                [BLACK] * 3 + [WHITE] * 4,
+                2,
+            )
+
+            pure_colors = (
+                "pure_red",
+                "pure_green",
+                "pure_blue",
+            )
+
+            pure_lines = named_lines_group(
+                3.2,
+                pure_colors,
+                pure_colors,
+                [BLACK, BLACK, WHITE],
+                6,
+            )
+
+            self.add(color_groups, other_lines, gray_lines, pure_lines)
+
+            VGroup(*self.mobjects).move_to(ORIGIN)
+
+
+The preferred way of using these colors is by importing their constants from manim:
+
+.. code-block:: pycon
+
+    >>> from manim import RED, GREEN, BLUE
+    >>> RED
+    '#FC6255'
+
+Note this way uses the name of the colors in UPPERCASE.
+
+Alternatively, you can also import this Enum directly and use its members
+directly, through the use of :code:`color.value`.  Note this way uses the
+name of the colors in lowercase.
+
+.. code-block:: pycon
+
+    >>> from manim.utils.color import Colors
+    >>> Colors.red.value
+    '#FC6255'
+
+.. note::
+
+    The colors of type "C" have an alias equal to the colorname without a letter,
+    e.g. GREEN = GREEN_C
+
+"""
+
+
+class ManimColor:
+    def __init__(
+        self,
+        value: str
+        | int
+        | list[int, int, int]
+        | list[int, int, int, int]
+        | list[float, float, float]
+        | list[float, float, float, float],
```

Comment:
Maybe using a ``TypeAlias` here as it is used in `ParsableManimColor` as well?

Reply:
I guess that would be a reasonable idea, but it's more readable with autocomplete and typehints like this 🤔

Reply:
I fixed that now with just using `ParsableManimColor` as parameter type in `ManimColor`